[READY]Adds cowboy boots!

Krysonism opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This PR adds 6 new types of cowboy boots. Brown, black and white cowboy boots have been added to the clothesmate. 

Two types of lizard skin boots can be crafted, these boots can then be exported for money.

An additional pair of fancy boots are currently unobtainable, but will be part of a future project.

Cowboy boots can now act as shelter for some type of snakelike creatures,  fill a pair with snakes as a prank or harebrained assassination plot.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Howdy partner! There has been some interest in adding cowboy hats for some time now, but what is a cowboy without his boots? This PR adds such boots and even some made of lizard skin, now finally those savage creatures are good for something, imagine that. Yeehaw!

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

![boots](https://user-images.githubusercontent.com/49783092/60390243-721fb100-9ad2-11e9-924b-cef86299887c.JPG)

## Changelog
:cl:
add: Cowboy boots have been added to the Clothesmate.
add: Lizard skin boots with decent export values have been added to the crafting menu.
add: Snakes, headslugs and alien larvae can now hide in cowboy boots.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Krysonism committed:
Merge remote-tracking branch 'upstream/master'

Krysonism committed:
Merge pull request #3 from tgstation/master

update fork

Krysonism committed:
Merge pull request #5 from tgstation/master

update master

Krysonism committed:
Merge pull request #7 from tgstation/master

update fork again

Krysonism committed:
Merge pull request #9 from tgstation/master

update fork

Krysonism committed:
Merge pull request #10 from tgstation/master

update fork

Krysonism committed:
Adds cowboy boots

optimumtact commented:
Two types of lizard skin boots can be crafted

:eyes:

optimumtact commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,47 @@
+	name = "lizard skin boots"
+	desc = "You can hear a faint hissing from inside the boots, you hope it is just a mournful ghost."
+	icon_state = "lizardboots_green"
+	armor = list("melee" = 10, "bullet" = 5, "laser" = 5, "energy" = 5, "bomb" = 5, "bio" = 0, "rad" = 0, "fire" = 0, "acid" = 0)
```
are you seriously adding armour to boots? 

stop

Krysonism replied:
I am certainly willing to remove the armour, the is precedent for armour on boots though.

 I figured it would be okay since the values were low enough and leg damage not being a big factor most fights.

nemvar replied:
They should probably have some fire and acid armor if any, because that effects the item more than the player wearing them.

Krysonism replied:
A little bit of fire resist makes sense.

Dorsisdwarf replied:
Which boots already have armour?

HomocidlePancakes commented:
Cowboy hat when?

Dawson1917 commented:
can i attach throwing stars to my boots to make chink chink sounds whenever i walk

Krysonism committed:
Removes armor and adds fire resist

Orange man say armor bad

MrPerson commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,46 @@
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 40, "acid" = 0) //lizards like to stay warm
+
+/obj/item/clothing/shoes/cowboy/lizard/masterwork
+	name = "Hugs-The-Feet lizard skin boots"
```
"\improper Hugs-The-Feet lizard skin boots"

Krysonism replied:
Will fix this tomorrow

MrPerson commented on code/modules/cargo/exports/gear.dm:
```diff
@@ -81,3 +81,19 @@
+	export_types = list(/obj/item/clothing/shoes/cowboy/lizard)
+	include_subtypes = FALSE
+
+datum/export/gear/lizardmasterwork
```
You're missing a leading / here.

Krysonism replied:
> 
> 
> can i attach throwing stars to my boots to make chink chink sounds whenever i walk

I am investigating this.

necromanceranne commented:
I can't believe cargonians are going to start skinning lizard people for cash.

PKPenguin321 commented:
I love this but do we still not have cowboy hats

ishitbyabullet commented:
Yeah, can this include a cowboy hat too? That'd sell the whole thing.

AarontheIdiot commented:
cowpoke

Krysonism commented:
> 
> 
> Yeah, can this include a cowboy hat too? That'd sell the whole thing.

possibly, if i can find out who was spriting the hats.

 I will be trying to implement spurs in the meantime(no promises)

GuyonBroadway commented:
Disappointed there's no mechanic for snakes/xeno larva to hide in boots.

ishitbyabullet commented:
Goonstation has a cowboy hat.

Krysonism commented:
The spur thing is probably beyond my abilities i tried to snowflake in shoe accessories that transferred a squeak component onto the shoes but i cant get it to work right.

Krysonism committed:
Minor fixes

YakumoChen commented:
Add 1% chance for boots to spawn a snake when created

Krysonism commented:
> 
> 
> Add 1% chance for boots to spawn a snake when created

All subtypes or only the non-crafted?

Krysonism committed:
Bootsnakes part 1

There feature we have all been waiting for, part one at least.

Krysonism commented:
![boot](https://user-images.githubusercontent.com/49783092/60411219-10437200-9bcc-11e9-9483-51a63fce0ac0.JPG)

Now i just need to add a way to scoop up additional snakes into the boots. 

If i let the mob resist out i could add support for scooping up alien larva and headslugs, we will see how productive i feel tomorrow.

Krysonism committed:
Bootsnakes part 2

Cowboy boots can now be filled with snakes, alien larvae and headslugs by click dragging the creature onto the boot. Sentient creatures can click drag themselves to enter and resist to get out.

bgobandit commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,88 @@
+
+/obj/item/clothing/shoes/cowboy
+	name = "cowboy boots"
+	desc = "A small sticker lets you know its been inspected for snakes, It is unclear how long ago the inspection took place..."
```
it's

Krysonism replied:
they?

Krysonism replied:
they've

Krysonism committed:
they've instead of its

wesoda25 commented:
ahhhhhhhhhh references

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,88 @@
+
+/obj/item/clothing/shoes/cowboy/lizard
+	name = "lizard skin boots"
+	desc = "You can hear a faint hissing from inside the boots, you hope it is just a mournful ghost."
```
comma splice

Krysonism replied:
I have changed out the comma for a semicolon. Do  you consider this acceptable? Perhaps you have a suggestion for a conjunction you'd like to see instead?

Thank you for reviewing my PR!

SpaceManiac replied:
That's fine.

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,88 @@
+	if(target.stat == DEAD)
+		return ..()
+	if(occupants.len >= max_occupants)
+		to_chat(user, "<span class='notice'>[src] seems to be full!</span>")
```
will read "The cowboy boots seems to be full"

Krysonism replied:
This will now read "the cowboy boot are full." nice and simple

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,88 @@
+/obj/item/clothing/shoes/cowboy/equipped(mob/living/carbon/user, slot)
+	. = ..()
+	if(slot == SLOT_SHOES)
+		if(occupants)
```
this `if` is unnecessary

Krysonism replied:
gone

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,88 @@
+			for(var/mob/living/occupant in occupants)
+				occupant.forceMove(get_turf(user))
+				to_chat(user, "<span class='userdanger'> You feel a sudden stabbing pain in your [pick("foot", "toe", "ankle")]!</span>",
+				"<span class='warning'>")
```
This `to_chat` should be combined into the `visible_message`.

Krysonism replied:
Done

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,88 @@
+				user.apply_damage(5, BRUTE, pick(BODY_ZONE_R_LEG, BODY_ZONE_L_LEG))
+				if(istype(occupant, /mob/living/simple_animal/hostile/retaliate/poison))
+					user.reagents.add_reagent(/datum/reagent/toxin, 4)
+				occupants -= occupant
```
if you're removing everything from the list, `.Cut()` it afterwards rather than removing each item one by one

Krysonism replied:
Done

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,88 @@
+	if(slot == SLOT_SHOES)
+		if(occupants)
+			for(var/mob/living/occupant in occupants)
+				occupant.forceMove(get_turf(user))
```
Should be `user.drop_location()` instead of `get_turf(user)`, otherwise it will pop out of lockers.

Krysonism replied:
Done

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,88 @@
+/obj/item/clothing/shoes/cowboy/container_resist(mob/living/user)
+	if(!do_after(user, 10, target = user))
+		return
+	user.forceMove(get_turf(src))
```
Ditto `drop_location`

Krysonism replied:
Done

Krysonism committed:
Fixes to make the maintainers happy

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,85 @@
+				user.reagents.add_reagent(/datum/reagent/toxin, 4)
+		occupants.Cut()
+
+/obj/item/clothing/shoes/cowboy/MouseDrop_T(mob/living/target, mob/user)
```
Needs to check that user is not incapacitated and is adjacent to the shoes

Krysonism replied:
I think i have fixed this.

SpaceManiac reviewed (changes requested):
Something I missed the first time

Krysonism committed:
More checks for snake loading +3u more toxin per snake

In the world of plasma cigars, I have decided that the snake boot assassination method is so hard to execute it should probably crit you without treatment.

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,85 @@
+/obj/item/clothing/shoes/cowboy/MouseDrop_T(mob/living/target, mob/living/user)
+	. = ..()
+	if(user.stat || !(user.mobility_flags & MOBILITY_USE) || user.restrained() || !Adjacent(user) || !user.Adjacent(target) || target.stat == DEAD)
+		return ..()
```
This line would end up with `..()` being called twice, which is *probably* harmless in this case, but still bad. Just `return` and the `.` calculated during `. = ..()` will be used.

Krysonism replied:
done

SpaceManiac commented on code/modules/clothing/shoes/miscellaneous.dm:
```diff
@@ -336,3 +336,85 @@
+		return ..()
+	if(occupants.len >= max_occupants)
+		to_chat(user, "<span class='notice'>[src] are full!</span>")
+		return ..()
```
Same

Krysonism replied:
done

SpaceManiac reviewed (changes requested):
Another thing I missed the first two times. Sorry.

Ledwin1233 commented:
18 naked cowboys at /tg/station!!

Krysonism committed:
Just return
